### DISCUSSION TITLE
ExpressionLanguage, evaluate float numbers.

### DISCUSSION THREAD

**Discussion by begicf** [symfony/symfony#48683] [Q&A] [answered]
### Symfony version(s) affected

6.2

### Description

Running this calculation operation return FALSE instead of TRUE:

`$ast = (new ExpressionLanguage())->evaluate('3084325.68==1838758.52 + 1245567.16');`

The sum of these two values ​​is 3084325.68 it is only happens with float numbers and if second decimal place ends with 8. 

 

### How to reproduce

`$ast = (new ExpressionLanguage())->evaluate('3084325.68==1838758.52 + 1245567.16');
var_dump($ast);`

Return false 

Correct is true

You need to run this part of the code, it returns FALSE instead of TRUE.

**Comment by AlexandreGerault** [comment, marked as answer]
I think it is not related to Symfony, but instead to the way float points are handled by computers. Reading [this article](https://0.30000000000000004.com/) might help you understanding what's going on here.

If you try comparing two floats number with `0.1 + 0.2 === 0.3`, you'll see it evaluates to `false`.

**Reply by begicf** [reply]
Thanks @AlexandreGerault, I managed to solve this problem by creating a custom rounding function, it only happens with some numbers.
